sisl/viz/plotly/input_fields/sisl_obj.py

```diff
@@ -44,16 +47,15 @@ def __init__(self, key, *args, **kwargs):
                 f'If there are multiple settings that accept {kwargs["dtype"]}, please use *_{valid_key}'
             )
 
-        super().__init__(key, *args, **kwargs)
-
 
 class GeometryInput(SislObjectInput):
 
+    dtype = (sisl.Geometry, "sile (or path to file) that contains a geometry")
     _dtype = (str, sisl.Geometry, *sisl.get_siles(attrs=['read_geometry']))
 
     def parse(self, val):
 
-        if isinstance(val, str):
+        if isinstance(val, (str, Path)):
             val = sisl.get_sile(val)
         if isinstance(val, sisl.io.Sile):
```

Comment:
oh, I found something here. Here you want to check for `sisl.io.BaseSile` since the `sisl.io.Sile` is only ascii siles.

That would enable you to also read from cdf files etc.

Reply:
Haven't you had problems reading geometries from non-ascii files?

Reply:
I did it ;)

Reply:
Thanks! I didn't know this